### ISSUE TITLE
Chromedriver patcher is broken for Chrome >= 115

### ISSUE THREAD

**Issue by Hyperz** [ultrafunkamsterdam/undetected-chromedriver#1402]
It looks like Google has deprecated https://chromedriver.storage.googleapis.com for Chrome versions over 114. As a result the patcher is throwing a 404 HTTPError with the latest Chrome update because https://chromedriver.storage.googleapis.com/LATEST_RELEASE_115 doesn't exist. There's a new API at https://googlechromelabs.github.io/chrome-for-testing/last-known-good-versions-with-downloads.json which replaces the old one.

Related Google announcement: https://groups.google.com/g/chromedriver-users/c/clpipqvOGjE/m/5NxzS_SRAgAJ

**Comment by ecossett-hc** [comment]
Thank you for posting this, @Hyperz. I am noticing the same issue. Also seeing this issue with an RPA software too. Kinda surprised I did not hear about that chromedriver change prior to now.

**Comment by jasonlecs** [comment]
How can this be solved?

**Comment by mdmintz** [comment]
According to https://github.com/SeleniumHQ/selenium/milestone/12, the earliest that the fix will be available in `selenium` is sometime next week.

Until then, you can use [SeleniumBase](https://github.com/seleniumbase/SeleniumBase)'s UC Mode as an alternative, which has the fix for Chrome 115.

First ``pip install seleniumbase``, and then run the following script with ``python``:

```python
from seleniumbase import Driver
import time

driver = Driver(uc=True)
driver.get("https://nowsecure.nl/#relax")
time.sleep(8)
driver.quit()
```

**Comment by Mangakk** [comment]
I change self.version_main = version_main to self.version_main = 114 in pather.py, all work, chrome version 115.0.5790.99

**Comment by ysfmrz** [comment]
I use your code (bypass is ok), but if click on some link in page, I see cloudflar block page!!!

**Comment by ysfmrz** [comment]
> I use your code (bypass is ok), but if click on some link in page, I see cloudflar block page!!!



> According to https://github.com/SeleniumHQ/selenium/milestone/12, the earliest that the fix will be available in `selenium` is sometime next week.
> 
> Until then, you can use [SeleniumBase](https://github.com/seleniumbase/SeleniumBase)'s UC Mode as an alternative, which has the fix for Chrome 115.
> 
> First `pip install seleniumbase`, and then run the following script with `python`:
> 
> ```python
> from seleniumbase import Driver
> import time
> 
> driver = Driver(uc=True)
> driver.get("https://nowsecure.nl/#relax")
> time.sleep(8)
> driver.quit()
> ```


I use your code (bypass is ok), but if click on some link in page, I see cloudflar block page!!!

**Comment by claell** [comment]
@mdmintz I checked that link to the Selenium milestone. But I didn't find an issue for this. Is there one that you can link? Seems like this is still not fixed on their side.

**Comment by mdmintz** [comment]
@claell Code has been merged, but not yet released: Relevant links:

* https://github.com/SeleniumHQ/selenium/pull/12208/files
* https://bugs.chromium.org/p/chromium/issues/detail?id=1466427#c27

**Comment by Suyogg668** [comment]
has this been fixed yet ??

**Comment by ecossett-hc** [comment]
I did confirm that patcher.py is setting the location to download the driver as "https://chromedriver.storage.googleapis.com" Which is the old location that only houses 114 and below. So unfortunately it seems to still be broken for me on Chrome 115 with the latest selenium (4.11.2) and the latest undetected_chromedriver (3.5.0)

**Comment by mdmintz** [comment]
@ecossett-hc Yep, the UC maintainer still needs to update: https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/1c704a71cf4f29181a59ecf19ddff32f1b4fbfc0/undetected_chromedriver/patcher.py#L27
First detect the Chrome version, then download from the correct location.

There's a patched fork in another repo with an example here: https://github.com/ultrafunkamsterdam/undetected-chromedriver/issues/1402#issuecomment-1644940051

**Comment by ecossett-hc** [comment]
@mdmintz Ah gotcha! I'll poke around with seleniumbase. Unfortunately our RPA software (based on Selenium) has hit issues with chrome 115 as well so I've been fighting both of these issues at once haha

**Comment by mdmintz** [comment]
@ecossett-hc For the seleniumbase formats that run with pytest, be sure to use `pytest --uc` to activate undetected-chromedriver mode.

**Comment by ecossett-hc** [comment]
@mdmintz will do!

**Comment by Suyogg668** [comment]
seems like the issue got resolved. its working with 115 for me

**Comment by BosterJack** [comment]
Same issue for me 
![Capture](https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/56207603/cc09fbd4-489b-4377-a18f-f301bf39ef84)

**Comment by itsciccio** [comment]
This issue still persists.

**Comment by itsciccio** [comment]
Found a temporary fix: https://github.com/ugorsahin/ChatGPT_Automation/issues/25#issuecomment-1688200265
